src/Categories/Category/Instance/Properties/Cats/Complete.agda

```diff
@@ -0,0 +1,113 @@
+{-# OPTIONS --without-K --safe #-}
+
+module Categories.Category.Instance.Properties.Cats.Complete where
+
+open import Data.Product
+open import Level
+open import Relation.Binary
+
+open import Categories.Category
+open import Categories.Category.Complete
+open import Categories.Category.Instance.Cats
+open import Categories.Diagram.Cone
+open import Categories.Morphism
+open import Categories.Morphism.Reasoning.Core
+open import Categories.NaturalTransformation
+open import Categories.NaturalTransformation.NaturalIsomorphism
+open import Categories.Functor
+
+-- TODO: figure out what I really want to be doing with levels
+Cats-Complete : (o ℓ e o′ ℓ′ e′ : Level) → Complete o ℓ e (Cats (o ⊔ ℓ) (o ⊔ ℓ) (o ⊔ ℓ))
+Cats-Complete o ℓ e o′ ℓ′ e′ {J} F = record
+  { terminal = record
+    { ⊤ = record
+      { N = record
+        -- this is a very simplified version of "NaturalTransformation (const One) F"
+        { Obj = Σ[ S ∈ (∀ j → Category.Obj (Functor.F₀ F j)) ] (∀ {j k} (f : J [ j , k ]) → _≅_ (Functor.F₀ F k) (Functor.F₀ (Functor.F₁ F f) (S j)) (S k))
+        ; _⇒_ = λ {(S₁ , _) (S₂ , _) → ∀ j → Functor.F₀ F j [ S₁ j , S₂ j ] }
```

Comment:
I'm a bit confused by this definition. Shouldn't we be requiring the compatibility condition for morphisms as well as for objects?

Reply:
That might be what I'm missing

Reply:
I've found that trying to simplify things often goes sideways, and that going with the full condition (in this case `NaturalTransformation (const One) F`) tends to end up being clearest in the end.

Reply:
> I'm a bit confused by this definition. Shouldn't we be requiring the compatibility condition for morphisms as well as for objects?

Looking at this again, I've figured out what the condition on morphisms wants to be. I'm rewriting it as private definitions rather than nested records and products, because it was getting a bit hairy to work on.